packages/core-utils/src/app/contract-deployment.ts

```diff
@@ -108,3 +117,28 @@ export const deploy = async (
 
   return deployContractFunction(wallet)
 }
+
+/**
+ * Returns an address's first deployed contract or `null` if that address hasn't
+ * created a contract yet. This is useful when creating a contract when a service
+ * is initialized and using that contract on subsequent runs.
+ * @param addresss The address that is being deployed from
+ * @returns contractAddress The address of the first deployed contract or `null` if one hasn't been deployed yet 
+ */
+export const getFirstDeployedContractAddress = async (
+  provider: Provider,
+  address: string,
+): Promise<string | null> => {
+  const nonce = 0
+
+  const contractAddress = getAddress(add0x(keccak256(RLP.encode([
+    getAddress(address),
+    stripZeros(hexlify(nonce))
+  ]))).substring(26));
```

Comment:
In general, it could be a good to create a util function for this since 1) there's a lot going on, and a function name / function comment could be useful and 2) It will likely be done elsewhere.